features/exploredash/src/main/java/org/dash/wallet/features/exploredash/data/AtmDao.kt

```diff
@@ -158,25 +253,45 @@ interface AtmDao : BaseDao<Atm> {
         types: List<String>,
         bounds: GeoBounds
     ): PagingSource<Int, Atm> {
-        return if (query.isNotBlank()) {
-            pagingSearch(
-                sanitizeQuery(query),
-                territory,
-                types,
-                bounds.northLat,
-                bounds.eastLng,
-                bounds.southLat,
-                bounds.westLng
-            )
+        return if (territory.isBlank() && bounds != GeoBounds.noBounds) {
+            // Search by coordinates (nearby) if territory isn't specified
+            if (query.isNotBlank()) {
+                pagingSearchByCoordinates(sanitizeQuery(query), types,
+                    bounds.northLat, bounds.eastLng, bounds.southLat, bounds.westLng)
+            } else {
+                pagingGetByCoordinates(types, bounds.northLat, bounds.eastLng, bounds.southLat, bounds.westLng)
+            }
+        } else {
+            // If location services are disabled or user picked a territory
+            // we search everything and filter by territory
+            if (query.isNotBlank()) {
+                pagingSearchByTerritory(sanitizeQuery(query), territory, types)
+            } else {
+                pagingGetByTerritory(territory, types)
+            }
+        }
+    }
+
+    suspend fun getPhysicalResultsCount(
```

Comment:
As much as I wanted to avoid that, there seems to be no good way to count pagination results without doing another request. The overhead is 10-100 ms depending on the density of the area, so it's not critical.